mino/minogrpc/server.go

```diff
@@ -476,42 +499,22 @@ func newOverlay(tmpl *minoTemplate) (*overlay, error) {
 		tokens:      tokens.NewInMemoryHolder(),
 		certs:       tmpl.certs,
 		router:      tmpl.router,
-		connMgr:     newConnManager(tmpl.myAddr, tmpl.certs, tmpl.useTLS),
+		connMgr:     newConnManager(tmpl.myAddr, tmpl.certs, tmpl.serveTLS),
 		addrFactory: tmpl.fac,
 		secret:      tmpl.secret,
 		public:      tmpl.public,
-	}
-
-	if tmpl.cert != nil && tmpl.useTLS {
-		chain := bytes.Buffer{}
-		for _, c := range tmpl.cert.Certificate {
-			chain.Write(c)
-		}
-
-		err := o.certs.Store(o.myAddr, chain.Bytes())
-		if err != nil {
-			return nil, xerrors.Errorf("failed to store cert: %v", err)
-		}
-	}
-
-	if tmpl.useTLS {
-		cert, err := o.certs.Load(o.myAddr)
-		if err != nil {
-			return nil, xerrors.Errorf("while loading cert: %v", err)
-		}
-
-		if cert == nil {
-			err = o.makeCertificate()
-			if err != nil {
-				return nil, xerrors.Errorf("certificate failed: %v", err)
-			}
-		}
```

Comment:
good call on moving all this stuff ahead of the overlay creation, it feels cleaner !